### PR TITLE
Move hunt creation/edit to a separate page

### PR DESCRIPTION
**Pull request by ebroder**
As we add more hunt configuration, we're rapidly outgrowing what's
reasonable to put into a relatively small modal. Given that, push it to
a separate page.

This is mostly pure copy-paste of the old modal content, but with a few
changes - namely handling breadcrumbs and URL parameters (instead of
having a hunt passed in), and slight changes to the post-submit flow.
Instead of closing the modal, we take different behavior depending on
whether we're creating a new hunt or editing an existing one. For
creation, we redirect to the newly created hunt. For edits, we scroll to
the bottom of the page and display a banner confirming that the edit was
successful, with a back button to take you back to the hunt list.

### CODE REVIEW

**Comment by ebroder** [comment]
For the record, I don't know that I'll want to keep the "single save button" design that we currently have as I start working through #630, since I think we may want to save each of the integrations separately. I opted to keep the behavior similar for now though and revisit later.

File: imports/client/components/HuntEditPage.tsx
Comment (zarvox) [inline imports/client/components/HuntEditPage.tsx:549]: While the previous modal form needed a way to dismiss it, since getting to this page now involves a navigation, I feel like between the browser back button and the breadcrumbs we don't need an additional way to do this.  This is no longer a modal form; it no longer needs its own "dismiss" button.  Can we take it out?

File: imports/client/components/HuntEditPage.tsx
Comment (zarvox) [inline imports/client/components/HuntEditPage.tsx:551]: Can we make this button align to the right edge of the form?  I think "bottom right" is where I see a majority of "submit" buttons in forms in general.

(And if there were multiple buttons, the default "do a useful thing" action should be the one aligned with the edge of the form, with other actions growing out towards the left as needed)

Comment (zarvox) [reply imports/client/components/HuntEditPage.tsx:551]: On reflection: I notice we're pretty inconsistent with "forms have save buttons on the right edge", and probably a big part of that is that the button is easy to lose on the far right if the page is very wide.  As a thought for the future: we might be well-served by picking a reasonable `max-width` to apply to some of our other forms and then moving the buttons to the right edge for them too.  It'd certainly help a bit with the visual balance; we tend to be very left-heavy at present (I'm thinking especially of OwnProfilePage and SetupPage).

File: imports/client/components/HuntEditPage.tsx
Comment (ebroder) [reply imports/client/components/HuntEditPage.tsx:549]: Haha. Honestly, I added the back button because I expected you to want the explicit navigation/nudge for post-save action. I don't mind dropping it

File: imports/client/components/HuntEditPage.tsx
Comment (ebroder) [reply imports/client/components/HuntEditPage.tsx:551]: Yeah, +1 - I'll wait until we land #767 and then incorporate its action button class.

**Comment by ebroder** [comment]
PTAL?

File: imports/client/components/HuntEditPage.tsx
Comment (zarvox) [inline imports/client/components/HuntEditPage.tsx:180]: As written, this results in there being no clickable breadcrumb for `/hunts/new`, which feels a little odd to me.

For the case where we're creating a new hunt, I think I'd lightly prefer if we had two breadcrumbs here -- one "Hunts", linking to `/hunts`, and then maybe "Create" (at `/hunts/new`).  Thoughts?

File: imports/client/components/HuntEditPage.tsx
Comment (zarvox) [inline imports/client/components/HuntEditPage.tsx:260]: Nice workflow optimization here :)

File: imports/client/components/HuntEditPage.tsx
Comment (zarvox) [inline imports/client/components/HuntEditPage.tsx:551]: Maybe label this button "Create" if we're in create mode and "Save" if we're in edit mode?

File: imports/client/components/HuntListPage.tsx
Comment (zarvox) [inline imports/client/components/HuntListPage.tsx:212]: I wonder if this would be better as a button with text "Create new hunt..." instead of an icon.

**Review by zarvox — APPROVED** [review]
Thanks, this is good!  My only comments are a few tiny nits that I'm not particularly attached to, so I'll approve and leave you the option of merging as-is or tweaking and submitting.

File: imports/client/components/HuntEditPage.tsx
Comment (ebroder) [reply imports/client/components/HuntEditPage.tsx:180]: I'll poke at this in a bit. The reason I didn't add a second breadcrumb initially is that you can't call hooks conditionally. I can try to modify `useBreadcrumb` to take `object | null` and not render the breadcrumb if null is passed in (similar to how `useSubscribe` works), but I'm not sure if that will actually work with `useBreadcrumb`'s weird state tracking.

Comment (ebroder) [reply imports/client/components/HuntEditPage.tsx:180]: OK managed to get this working by just rearranging the route map a bit. I'll push in a sec

**Pull request merged**
